tests/rest_api/test_0004_tasks.py

```diff
@@ -0,0 +1,181 @@
+        exclude_paths = [f"root[{i}]['assignee']" for i in range(len(filtered))]
+        exclude_paths.extend([f"root[{i}]['updated_date']" for i in range(len(filtered))])
+
+        self._test_can_see_project_tasks(self._SAND_PROJ_OWNER, self._SAND_PROJ_ID, filtered, exclude_paths)
+        self._test_can_see_project_tasks(user1['username'], self._SAND_PROJ_ID, filtered, exclude_paths)
+
+        # check user2 is able to see a task that was assigned to him
+        response = get_method(user2['username'], f'tasks')
+        response_data = response.json()
+        assert response.status_code == HTTPStatus.OK
+        assert any(task['id'] == self._SAND_TASK_ID for task in response_data['results'])
+
+    # [organization] maintainer can see task data even if he has no ownerships in corresponding Project, Task
+    def test_organization_maintainer_can_see_project_tasks(self, tasks):
+        filtered_tasks = list(filter(lambda x: x['project_id'] == self._ORG_PROJ_ID, tasks))
+        self._test_can_see_project_tasks(self._ORG_MAINTAINER, self._ORG_PROJ_ID, filtered_tasks, org=self._ORG)
+
+    # [organization] supervisor cannot see task data if he has no ownerships in corresponding Project, Task
+    def test_organization_supervisor_cannot_see_project_tasks(self):
+        self._test_cannot_see_project_tasks(self._ORG_SUPERVISOR, self._ORG_PROJ_ID, org=self._ORG)
+
+    # [organization] worker (as role) that has one of these ownerships: [Owner, Project:owner, Assignee, Project:assignee], can see task data
+    def test_organization_entitled_worker_can_see_project_tasks(self, users, tasks):
+        # Worker can't create tasks/projects, so we only check assignee
+        worker2 = list(filter(lambda x: x['username'] == 'worker2', users))[0]
+        worker3 = list(filter(lambda x: x['username'] == 'worker3', users))[0]
+        self._assign_user_to_proj(worker2, self._ORG_PROJ_ID, org=self._ORG)
+        self._assign_user_to_task(worker3, self._ORG_TASK_ID, org=self._ORG)
+        filtered = list(filter(lambda x: x['project_id'] == self._ORG_PROJ_ID, tasks))
+        exclude_paths = [f"root[{i}]['assignee']" for i in range(len(filtered))]
+        exclude_paths.extend([f"root[{i}]['updated_date']" for i in range(len(filtered))])
+
+        self._test_can_see_project_tasks('worker2', self._ORG_PROJ_ID, filtered, exclude_paths)
+
+        # check worker3 is able to see a task that was assigned to him
+        response = get_method(worker3['username'], f'tasks')
+        response_data = response.json()
+        assert response.status_code == HTTPStatus.OK
+        assert any(task['id'] == self._ORG_TASK_ID for task in response_data['results'])
+
+
+class TestPostTasks:
+    _ADMIN = 'admin1'
+    _SAND_PROJ_ID = 1
+    _SAND_PROJ_OWNER = 'business1'
+    _SAND_PROJ_NOT_OWNER = 'business2'
+    _ORG = 'org2'
+    _ORG_PROJ_ID = 2
+    _ORG_SUPERVISOR_NOT_OWNER = 'user3'
+    _ORG_WORKER_NO_OWNERSHIP = 'worker3'
```

Comment:
Consider getting rid of hardcoded names and ids in tests. Avoiding hard-coded usernames, identifiers, etc. makes it easier to make new changes to the test database.

Fixture `find_user` that allow easy to find needed user with specific organization, role, and privilege:
```
user0 = find_users(org=1, role='maintainer')[0]['username']
user1 = find_users(privilege='user', org=2)[0]['username']
```

[Here](https://github.com/openvinotoolkit/cvat/blob/377fe2cfabf6d3f88f74afa9a49338253ec75543/tests/rest_api/conftest.py#L135-L141) possible solution (not merged yet) for searching needed user with specific ownership for the project

Reply:
@kirill-sizov Updated. Now the PR depends on your PR #4351